frontend/public/components/alert.tsx

```diff
@@ -95,13 +97,22 @@ class MonitoringPageWrapper extends SafetyFirst<AlertsPageWrapperProps, null> {
     }
 
     poll(`${prometheusBaseURL}/api/v1/rules`, 'rules', data => {
-      // Flatten the rules data to make it easier to work with and also discard non-alerting rules since those are
-      // the only ones we will be using
-      const allRules = _.flatMap(_.get(data, 'groups'), 'rules');
-      const alertingRules = _.filter(allRules, {type: 'alerting'});
+      // Flatten the rules data to make it easier to work with, discard non-alerting rules since those are the only ones
+      // we will be using and add a unique ID to each rule.
+      const groups = _.get(data, 'groups');
+      const rules = _.flatMap(groups, g => {
+        const addID = r => {
+          const key = [g.file, g.name, r.name, r.duration, r.query, ..._.map(r.labels, (k, v) => `${k}=${v}`)].join(',');
```

Comment:
Are these fields mutable? What happens when one changes?

Reply:
Unfortunately, if any of these key fields change (e.g. if someone changes the `query`), `id` will change and the page will switch to a "Not Found" message. That is a disadvantage of this approach.